File: src/java/org/waxeye/parser/WildCardTransition.java
/*
 * Waxeye Parser Generator
 * www.waxeye.org
 * Copyright (C) 2008-2010 Orlando Hill
 * Licensed under the MIT license. See 'LICENSE' for details.
 */
package org.waxeye.parser;

import org.waxeye.ast.IAST;

/**
 * A transition cost of matching any character.
 *
 * @param <E> The AST type.
 *
 * @author Orlando Hill
 */
public final class WildCardTransition <E extends Enum<?>> implements ITransition<E>
{
    /** {@inheritDoc} */
    public IAST<E> acceptVisitor(final ITransitionVisitor<E> visitor)
    {
        return visitor.visitWildCardTransition(this);
    }
}

File: src/php/parser/transition/WildcardTransition.php
<?php


namespace parser\transition;


use ast\Char;
use ast\IAST;

class WildcardTransition implements ITransition
{
    public function visitTransition(string $input, int $position): ?IAST
    {
        $result = substr($input, $position, 1);

        if ($result !== false) {
            return new Char($input[$position], $position, "char");
        } else {
            return null;
        }
    }

    /**
     * @inheritDoc
     */
    public function jsonSerialize()
    {
        return array("WildcardTransition" => get_object_vars($this));
    }

    public function __toString()
    {
        return json_encode($this);
    }
}
